core/util/src/main/java/org/eclipse/rdf4j/common/iteration/Iteration.java

```diff
@@ -44,11 +45,22 @@
 	/**
 	 * Removes from the underlying collection the last element returned by the iteration (optional operation). This
 	 * method can be called only once per call to next.
-	 * 
+	 *
 	 * @throws UnsupportedOperationException if the remove operation is not supported by this Iteration.
 	 * @throws IllegalStateException         If the Iteration has been closed, or if <tt>next()</tt> has not yet been
 	 *                                       called, or <tt>remove()</tt> has already been called after the last call to
 	 *                                       <tt>next()</tt>.
 	 */
 	public void remove() throws X;
+
+	/**
+	 *
+	 * Convert the results to a Java 8 Stream. If this iteration implements CloseableIteration it should be closed (by
+	 * calling Stream#close() or using try-with-resource) if it is not fully consumed.
+	 *
+	 * @return stream
+	 */
+	default Stream<E> stream() {
+		return Iterations.stream(this);
+	}
```

Comment:
@jeenbroekstra 

The javadocs say that most streams don't need to be closed:

> Streams have a BaseStream.close() method and implement AutoCloseable, but nearly all stream instances do not actually need to be closed after use. Generally, only streams whose source is an IO channel (such as those returned by Files.lines(Path, Charset)) will require closing. Most streams are backed by collections, arrays, or generating functions, which require no special resource management. (If a stream does require closing, it can be declared as a resource in a try-with-resources statement.)

I have removed the duplicate method and modified the docs (see above) to make it up to the user to decide what they want to do.